src/MeshKernelNET/Api/IMeshKernelApi.cs

```diff
@@ -1172,9 +1172,27 @@ public interface IMeshKernelApi : IDisposable
         /// <param name="disposableGeometryListIn">The input samples</param>
         /// <param name="interpolationParameters">The settings for the interpolation algorithm</param>
         /// <param name="samplesRefineParameters">The settings for the interpolation related to samples</param>
-        /// <returns>If the operation succeeded</returns>
+        /// <returns>Error code</returns>
         int Mesh2dRefineBasedOnSamples(int meshKernelId, in DisposableGeometryList disposableGeometryListIn,
                                        double relativeSearchRadius, int minimumNumSamples, in MeshRefinementParameters meshRefinementParameters);
+        /// <summary>
+        /// Rotates a mesh2d about a given point by a given angle
+        /// </summary>
+        /// <param name="meshKernelId">Id of the grid state</param>
+        /// <param name="centreX">X-coordinate of the centre of rotation</param>
+        /// <param name="centreY">Y-coordinate of the centre of rotation</param>
+        /// <param name="angle">Angle of rotation</param>
+        /// <returns>Error code</returns>

```

Comment:
angle in radians I suppose? suggest to add it to the documentation.

Reply:
Degrees actually. I'll add units everywhere.